docs/workflows/executions/manual-partial-and-production-executions.md

```diff
@@ -0,0 +1,54 @@
+---
+#https://www.notion.so/n8n/Frontmatter-432c2b8dff1f43d4b1c8d20075510fe4
+title: Manual, partial, and production executions
+description: How manual, partial, and automatic workflow executions differ.
+contentType: explanation
+---
+
+# Manual, partial, and production executions
+
+There are some important differences in how n8n executes workflows manually (by clicking the **Test Workflow** button) and automatically (when the workflow is **Active** and triggered by an event or schedule).
+
+## Manual executions
+
+Manual executions allow you to run workflows directly from the [canvas](/glossary.md#canvas-n8n) to test your workflow logic. These executions are "ad-hoc" — they run only when you manually select the **Test workflow** button.
+
+Manual executions make building workflows easier by allowing you to iteratively test as you go, following the flow logic and seeing data transformations. You can test conditional branching, data formatting changes, and loop behavior by providing different input items and modifying node options.
+
+/// note | Pinning execution data
+When performing manual executions, you can use [data pinning](/data/data-pinning.md) to "pin" or "freeze" the output data of a node. On future runs, instead of executing the pinned node, n8n will substitute the pinned data and continue following the flow logic. This allows you to iterate without operating on variable data or repeating queries to external services. Production executions ignore all pinned data.
```

Comment:
I don't want to go overboard on this note, but it might be worth briefly mentioning editing pinned data (https://docs.n8n.io/data/data-editing/) too.

Reply:
No, I think that's a great call. Adding that and a break to try to maintain readability.

```suggestion
When performing manual executions, you can use [data pinning](/data/data-pinning.md) to "pin" or "freeze" the output data of a node. You can optionally [edit the pinned data](https://docs.n8n.io/data/data-editing/) as well.

On future runs, instead of executing the pinned node, n8n will substitute the pinned data and continue following the flow logic. This allows you to iterate without operating on variable data or repeating queries to external services. Production executions ignore all pinned data.
```